app/services/amr/csv_upserter.rb

```diff
@@ -0,0 +1,66 @@
+require 'upsert'
+require 'upsert/active_record_upsert'
+require 'upsert/connection/postgresql'
+require 'upsert/connection/PG_Connection'
+require 'upsert/merge_function/PG_Connection'
+require 'upsert/column_definition/postgresql'
+
+module Amr
+  class CsvUpserter
```

Comment:
This just does upserting now - possibly the parser could create an array of hashes with the relevant fields already set up rather than doing it here. Left it as is for now, but kind of feels like it would be cleaner for this not to need to now about the config...

Reply:
Yes - so a third part of the process to sit between the Parser and the Upserter - Validation & massgaging in to a hash or VO which can then be passed on